File: frontend/src/types/question.ts
export interface MCQuestion {
  id: number;
  document_id: number;
  nos_unit_id: number | null;
  criterion_id: number | null;
  question_text: string;
  option_a: string;
  option_b: string;
  option_c: string;
  option_d: string;
  correct_option: 'A' | 'B' | 'C' | 'D';
  explanation: string;
  source_page_reference: string | null;
  difficulty_level: 'easy' | 'medium' | 'hard';
  is_duplicate: boolean;
  created_at: string;
  updated_at: string;
  tags: QuestionTag[];
}

export interface QuestionTag {
  id: number;
  name: string;
}

export interface GenerationJob {
  id: number;
  document_id: number;
  status: 'pending' | 'generating' | 'completed' | 'failed';
  total_criteria: number;
  processed_criteria: number;
  error_message: string | null;
  started_at: string | null;
  completed_at: string | null;
}

File: backend/app/schemas/question.py
from datetime import datetime

from pydantic import BaseModel


class TagResponse(BaseModel):
    id: int
    name: str

    model_config = {"from_attributes": True}


class QuestionResponse(BaseModel):
    id: int
    document_id: int
    nos_unit_id: int | None
    criterion_id: int | None
    question_text: str
    option_a: str
    option_b: str
    option_c: str
    option_d: str
    correct_option: str
    explanation: str
    source_page_reference: str | None
    difficulty_level: str
    is_duplicate: bool
    created_at: datetime
    updated_at: datetime | None
    tags: list[TagResponse] = []

    model_config = {"from_attributes": True}


class QuestionListResponse(BaseModel):
    questions: list[QuestionResponse]
    total: int
    page: int
    per_page: int


class QuestionUpdate(BaseModel):
    question_text: str | None = None
    option_a: str | None = None
    option_b: str | None = None
    option_c: str | None = None
    option_d: str | None = None
    correct_option: str | None = None
    explanation: str | None = None
    difficulty_level: str | None = None


class TagCreate(BaseModel):
    name: str


class QuestionStatsResponse(BaseModel):
    total: int
    by_difficulty: dict[str, int]
    by_document: dict[str, int]
    duplicates_count: int
